app/GraphQL/Mutations/Login.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\User;
+use GraphQL\Error\Error;
+use Illuminate\Support\Facades\Auth;
+
+class Login
+{
+    /**
+     * @param  null  $_
+     * @param  array<string, mixed>  $args
+     */
+    public function __invoke($_, array $args): User
+    {
+        $guard = Auth::guard();
+
+        if( ! $guard->attempt($args)) {
```

Comment:
Probably not a good idea to use `$args` as-is. Right now, it implicitly depends on the shape of the mutation args (User model fields might be something different than 'email' and 'password').

```php
$credentials = collect($args)->only('email', 'password')->all();
```

Reply:
I don't think it is necessary to future-proof against potential future renamings. Also, your implementation would also break if either the argument names or the model fields changed.

Reply:
Yeah, leave it out for the sake of brevity. Not that important for the example anyways.